besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2811,12 +2806,17 @@ private SynchronizerConfiguration buildSyncConfig() {
   }
 
   private TransactionPoolConfiguration buildTransactionPoolConfiguration() {
-    final var stableTxPoolOption = stableTransactionPoolOptions.toDomainObject();
-    return ImmutableTransactionPoolConfiguration.builder()
-        .from(stableTxPoolOption)
-        .unstable(unstableTransactionPoolOptions.toDomainObject())
-        .saveFile((dataPath.resolve(stableTxPoolOption.getSaveFile().getPath()).toFile()))
-        .build();
+    final var txPoolConf = transactionPoolOptions.toDomainObject();
+    final var txPoolConfBuilder =
+        ImmutableTransactionPoolConfiguration.builder()
+            .from(txPoolConf)
+            .saveFile((dataPath.resolve(txPoolConf.getSaveFile().getPath()).toFile()));
+
+    if (getActualGenesisConfigOptions().isZeroBaseFee()) {
```

Comment:
Does this mean that you need to be on the `london` fork in order to get the new behaviour? Could it be made to work in the same way for `--min-gas-price=0` for users pre-london?

Reply:
I thought about that, but setting `--min-gas-price=0` does not necessarily imply that it is a zero-gas-price network (actually also in zeroBaseFee network, the maxGasPrice could be > 0).

So what if we introduce a specific option, that acts like a _profile_, that state clearly that we want a _gas free_ network, for example `--gas-free`? that has the effect of setting sound values for other options like the `min-gas-price` and the `tx-pool-price-bump`, instead of trying to deduce the intent of the network from single options.

what do you think of this approach?

Reply:
In principle I agree @fab-10. However, I do have a couple of concerns about introducing it under this PR:

1. Most people are probably using `--min-gas-price=0` and/or `zeroBaseFee: true` to create what they think of as a `gas free` network. I agree it's not strictly gas free - but it probably means that defaulting `--tx-pool-price-bump` to `0` if you're using one/both of those settings is what most people would expect. They can after all set `--tx-pool-price-bump= (num > 0)` if they don't want it
2. I think if we had a `--gas-free` option people would have additional expectations if they set it - not just a default price bump of `0`. For example, would allocating initial funds to accounts be valid if `--gas-free` was set? Would setting a transaction with a `gasPrice>0` be allowed?

So I agree that `--gas-free` would be a really useful option. But I think it's probably worth separate discussion & PR to introduce it.

Reply:
Ok I see your point about `min-gas-price` and will implement your suggestion.

And let's continue the discussion about a `gas-free` profile, and all the feature that it should enable separately.